Detectors/TRD/base/src/Digit.cxx

```diff
@@ -11,35 +11,46 @@
 #include "TRDBase/Digit.h"
 namespace o2::trd
 {
+
+using namespace constants;
+
 Digit::Digit(const int det, const int row, const int pad, const ArrayADC adc)
 {
   setDetector(det);
   setROB(row, pad);
   setMCM(row, pad);
   setADC(adc);
+  setChannel(NADCMCM - 2 - (pad % NCOLMCM));
 }
+
 Digit::Digit(const int det, const int row, const int pad) // add adc data in a seperate step
```

Comment:
I wonder if we want a separate constructor to help instantiate shared (hidden) pads. This comes back to my comment above how we create digits in MC events.

Reply:
Well this starts to beg the question, what actually is a digit.
Is it a signal received on a pad or a series of timebins accumulated in the mcm.
I fear certain people will give the one answer and others the other.

Reply:
I think even one person might give one answer now, and another one at some other time.

The question is "what is an o2::trd::Digit?", and the code (the internal data members of the class) suggests that it is the signal, with up to 30 timebins, recorded by one ADC.

Reply:
I concur.

Reply:
So will we in the end have at max 144 `o2::trd::Digit`s per pad row of one detector or 168 which would include the 8x3 digits on shared pads? If I understand correctly we favor the latter, right? That would be an increase of the digit size of 17%. I am not sure if that is feasible?

Reply:
I do not think that data volume is an issue here. I do not see a need to store digits in a regular MC simulation - we would only keep the tracklets.

In any case, the "splitting" of shared pads is also done in data, so it is consistent to do it this way.

Reply:
Ok fine with me. So what do we need the function isSharedDigit for? What would be a use case? If I would convert digits from AliRoot to O2 for example I would simply check during the creation of the O2 digits whether or not I actually need to create one or two digits.

Reply:
due to the trap simulator being pad/row organise (hence the 8 trapsims instantiated) it would lead to a double write of the contents. It would be better to keep the digits single as in AliRoot and then the trapsim can sort out where the timebins are assigned.

Reply:
Where would be a double write? I don't see the function being used anywhere. If we keep max 21 digits per MCM we should be save, right? We could also make a short meeting on zoom to discuss the digits. I am free this afternoon I think

Reply:
I agree re chat. We have meeting in 20 minutes.

Reply:
So I think this is resolved. We have to fix TRAPsim, and @jolopezl should check that the digitizer produces two digits for shared pads.